src/Apps/Settings/Routes/SavedSearchAlerts/Components/SavedSearchAlertEditForm.tsx

```diff
@@ -172,9 +185,18 @@ const SavedSearchAlertEditForm: React.FC<SavedSearchAlertEditFormProps> = ({
                 <Flex flexWrap="wrap" mx={-0.5}>
                   <SavedSearchAlertPills
                     items={pills}
-                    onDeletePress={removePill}
+                    onDeletePress={handleRemovePillPress}
                   />
                 </Flex>
+
+                <Separator my={1} />
+
+                <PriceRangeFilter
+                  onChange={updatePriceRange}
+                  priceRange={criteria.priceRange ?? DEFAULT_PRICE_RANGE}
```

Comment:
**suggestion**: keep this logic inside the `PriceRangeFilter`. Something like this

```javascript
export const PriceRangeFilter = () => {
  const { criteria, addCriteriaValue } = useSavedSearchAlertContext()

  const handlePriceRangeUpdate = (updatedRange: CustomRange) => {
    addCriteriaValue("priceRange", updatedRange.join("-"))
  }

  return (
    <Expandable
      label="Set price range you are interested in"
      expanded
      borderColor="white100"
    >
      <PriceRange
        priceRange={criteria.priceRange ?? DEFAULT_PRICE_RANGE}
        onPriceRangeUpdate={handlePriceRangeUpdate}
      />
    </Expandable>
  )
}
```